apis/python/src/tiledbsoma/_exception.py

```diff
@@ -55,6 +55,46 @@ def is_does_not_exist_error(e: tiledb.TileDBError) -> bool:
     return False
 
 
+class AlreadyExistsError(SOMAError):
+    """Raised when attempting to create an already existing SOMA object.
+
+    Lifecycle: experimental
+    """
+
+    pass
+
+
+def is_already_exists_error(e: tiledb.TileDBError) -> bool:
+    """Given a TileDBError, return true if it indicates the object already exists
+
+    Lifecycle: experimental
+
+    Example:
+        XXX EDIT ME
```

Comment:
intended?

Reply:
😬 

Derp, I put up this PR too quick -- my apologies